### DISCUSSION TITLE
Scheduler Task not right timing

### DISCUSSION THREAD

**Discussion by DasLixou** [Minestom/Minestom#1083] [Help] [answered]
Hello! i build a task with SchedulerManager and have a repeat with `Duration.ofMillis(1000/20)` which is 50 millis so 20 times a sec. But when counting up 1 every time and printing every 20, you see that it only counts up by 1 like 7-9 times a sec, so it prints 20 only 4-5 secs. (its defenetly too slow). But: when i change it to `Duration.ofNanos(1)` its very fast as expected.

What have i done wrong? Is this a bug with SchedulerManager?

**Comment by TheMode** [comment, marked as answer]
Sync tasks in the scheduler are processed on a per-tick basis, meaning that there is no precise guarantee on your `Duration`.

In this case you should either make your task Async (may be possible depending on what it does), or use `TaskSchedule.nextTick()`
